sample-dapps/ethereum-wallet-explorer/src/app/api/wallet/transactions/route.ts

```diff
@@ -0,0 +1,40 @@
+import { ethers } from 'ethers';
+import { handleError, parseAndValidateAddress } from '../../utils/helpers';
+
+export async function GET(request: Request) {
+    const url = new URL(request.url);
+    let internalFlag;
+    const walletAddress = url.searchParams.get('walletAddress');
+    try {
+        const validAddress = await parseAndValidateAddress(walletAddress);
+        const provider = new ethers.JsonRpcProvider(process.env.ETHEREUM_RPC_URL);
+        const appearances = await provider.send("tb_getAppearances", [{"address": validAddress}]);
+        const txnsData = await appearances.data
+        const transactions = [];
+        for (let txn of txnsData) {
```

Comment:
I'm not entirely sure how these methods above work, but would you want to add a `limit` here to prevent doing potentially hundreds or thousands of loops?

Reply:
added a slice() to the array for a hard limit to the last 100